### PR TITLE
Fix timezone issue in materialize & materialize_incremental

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: Fixes an issue of mixing tz-aware and tz-naive timestamps. This was happening in fs.materialize & fs.materialize_incremental (both call _materialize_single_feature_view where I applied the fix), in FeatureView where the time zones aren't persisted between to_proto & from_proto, and in LocalSqlite:pull_latest_from_table_or_query where the source dataframe could have some tz-naive timestamps.

I also refactored `test_materialize_from_bigquery_to_datastore.py` into `test_materialize.py` where I maintain both existing test cases but also add a unit test that runs in the local mode. Lastly, I made sure to mix in tz-naive and tz-aware timestamps so that we make sure this bug is now well-tested.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fixes tz-naive / tz-aware bug with materialization
```

### CODE REVIEW

File: sdk/python/tests/test_materialize.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                online_store=OnlineStoreConfig(
+                    local=LocalOnlineStoreConfig(
+                        path=str(Path(data_dir_name) / "online_store.db")
+                    )
+                ),
+            )
+            fs = FeatureStore(config=config)
+            fs.apply([fv])
+
+            self.run_materialization_test(fs, fv)
```
Comment (woop) [inline sdk/python/tests/test_materialize.py:168]: This is a bit of a nitpick, but what do you think of restructuring the tests a little bit? I dont really see the need for a class and methods here when we can use functions. I'm worried that the class will create unnecessary scoping or tests will evolve to influence each other over time. I also see you are mixing graph building in a push (`self.run_materialization_test(fs, fv)`) and pull (`@pytest.mark.parametrize`) manner. We should probably just pick one of the two, and the pull based graph building seems cleaner to me.

something like

def other_functions()

def prep_bq_fs()
  do all bq setup

def prep_local_fs()
  do all local setup

@pytest.mark.parametrize( depends on prep_bq_fs and prep_local_fs)
def test_materialization()

**Review by woop — CHANGES_REQUESTED** [review]
Mostly looks good, just some comments on the test structure

File: sdk/python/tests/test_materialize.py (hunk 1 above)
Comment (tsotnet) [reply sdk/python/tests/test_materialize.py:168]: Most of this makes sense, but the problem with
> @pytest.mark.parametrize( depends on prep_bq_fs and prep_local_fs)
def test_materialization()

is that I either have to make the entire test function as integration test or unit test. There's no way to selectively mark prep_bq_fs as integration and prep_local_fs as unit test (I just tried this locally).

Comment (tsotnet) [reply sdk/python/tests/test_materialize.py]: @woop I went ahead with the push based approach. Basically make everything functions that we call in order to not duplicate anything

**Review by woop — APPROVED** [review]

**Pull request merged**